test/cpp/naming/resolver_component_test.cc

```diff
@@ -367,5 +360,50 @@ int main(int argc, char **argv) {
   ::testing::InitGoogleTest(&argc, argv);
   grpc::testing::InitTest(&argc, &argv, true);
 
-  return RUN_ALL_TESTS();
+  grpc_init();
+
+  SubProcess *dns_server_subprocess = NULL;
+
+  if (FLAGS_start_local_dns_server == true) {
+    /* spawn a DNS server subprocess*/
+    local_dns_server_port = grpc_pick_unused_port_or_die();
+    std::string my_bin = argv[0];
+    std::string bin_dir = my_bin.substr(0, my_bin.rfind('/'));
+    std::vector<std::string> args = {
+        "tools/run_tests/python_utils/dns_server.py",
+        "--dns_port=" + std::to_string(local_dns_server_port)};
+    dns_server_subprocess = new SubProcess(args);
+
+    // Wait for the DNS server to stand up.
+    // TODO: can we get rid of the need to sleep here?
+    // Without this sleep, some polling engines timeout and others
+    // fail fast.
+    gpr_sleep_until(gpr_time_add(gpr_now(GPR_CLOCK_REALTIME),
+                                 gpr_time_from_seconds(1, GPR_TIMESPAN)));
+    gpr_log(GPR_INFO, "started local DNS server subprocess: |%s %s|",
```

Comment:
Maybe also log the pid, if there's an easy way to get at it?

Reply:
I couldn't see an easy way to do this. Even with `gpr_subprocess` directly, the `pid` info appears to be platform-specific and not exposed. Using fork/exec directly I could, but I'd lean towards `gpr_subprocess` and its `SubProcess` (I think this wrapper is more portable too)